Objective-C/CBLReplicator.mm

```diff
@@ -548,40 +586,65 @@ static void onDocsEnded(C4Replicator *repl,
 
 
 - (void) onDocsEnded: (NSArray<CBLReplicatedDocument*>*)docs pushing: (BOOL)pushing {
-    for (CBLReplicatedDocument* doc in docs) {
+    NSMutableArray* posts = [NSMutableArray array];
+    for (CBLReplicatedDocument *doc in docs) {
         C4Error c4err = doc.c4Error;
-        if (!c4err.code) {
-            continue;
-        }
-        
         if (!pushing && c4err.domain == LiteCoreDomain && c4err.code == kC4ErrorConflict) {
-            // Conflict pulling a document -- the revision was added but app needs to resolve it
-            CBLLogInfo(Sync, @"%@: pulled conflicting version of '%@'", self, doc.id);
-            NSError* error;
-            
-            if ([_config.database resolveConflictInDocument: doc.id
-                                       withConflictResolver: _config.conflictResolver
-                                                      error: &error]) {
-                [doc resetError]; // Reset the error as successfully resolving the conflicts
-            } else {
-                CBLWarn(Sync, @"%@: Conflict resolution of '%@' failed: %@",
-                        self, doc.id, error);
-                [doc updateError: error];
-            }
+            [self resolveConflict: doc];
+        } else {
+            [posts addObject: doc];
+            [self logErrorOnDocument: doc pushing: pushing];
         }
-        
-        if (doc.c4Error.code)
-            CBLLogInfo(Sync, @"%@: %serror %s '%@': %d/%d", self, (doc.isTransientError ? "transient " : ""),
-                       (pushing ? "pushing" : "pulling"), doc.id, c4err.domain, c4err.code);
     }
-    
+    if (posts.count > 0)
+        [self postDocumentReplications: posts pushing: pushing];
+}
+
+
+- (void) resolveConflict: (CBLReplicatedDocument*)doc {
+    _conflictCount++;
+    dispatch_async(_conflictQueue, ^{
+        [self _resolveConflict: doc];
+        CBL_LOCK(self) {
+            if (--_conflictCount == 0 && _deferStoppedNotification) {
+                Assert(_rawStatus.level == kC4Stopped);
+                Assert(_state == kCBLStateStopping);
+                [self stopped];
+            }
+        }
+    });
+}
+
+
+- (void) _resolveConflict: (CBLReplicatedDocument*)doc {
+    CBLLogInfo(Sync, @"%@: Resolve conflicting version of '%@'", self, doc.id);
+    NSError* error;
+    if ([_config.database resolveConflictInDocument: doc.id
+                               withConflictResolver: _config.conflictResolver
+                                              error: &error])
+        [doc resetError];
+    else
+        CBLWarn(Sync, @"%@: Conflict resolution of '%@' failed: %@", self, doc.id, error);
```

Comment:
[doc updateError: error] is not happening. this will cause all errors to report as CBLConflict. Even when the document is not found.

Reply:
Will fix this.